src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -67,7 +75,8 @@ public void onCompletion(RecordMetadata md, Exception e) {
 public class MaxwellKafkaProducer extends AbstractProducer {
 	static final Object KAFKA_DEFAULTS[] = {
 		"compression.type", "gzip",
-		"metadata.fetch.timeout.ms", 5000
+		"metadata.fetch.timeout.ms", 5000,
+		"retries", 1
```

Comment:
mostly because of `NotLeaderForPartition` problems.  Been seeing long GC pauses result in a node getting kicked out of leading partitions, which means that the first try to write the records results in a `NotLeaderForPartiton` error, which is not retried by default.  So the metadata is updated, but the row is senselessly lost. 

But I think you also misunderstand the semantics of `retries=1` -- if the second delivery attempt succeeds, it will call the completion handler as normal -- in which case we'll checkpoint the offset and continue.  If it fails, it should call the completion handler with an exception, at which point we'll crash and hopefully transmit the message when we restart.  

Maxwell should never drop the message and keep going without checkpointing -- see the change in the `onCompletion` handler to either willfully drop the message, or just `throw new RuntimeException`